File: ksb-immo/src/main/java/sn/ksb/immo/ksbimmo/application/models/Garant.java
package sn.ksb.immo.ksbimmo.application.models;

import lombok.*;
import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.Type;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import java.util.UUID;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
@Builder
public class Garant {
    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @Type(type = "uuid-char")
    private UUID id;

    private String nom;

    private String prenom;

    private String adresse;

    @Column(unique = true)
    private String telephone;

    @Column(unique = true)
    private String email;

    private String profession;

    @Column(unique = true)
    private String cni;
}

File: ksb-front/src/app/shared/models/garant.ts
import { SituationProfessionnelle } from "./situation-professionnelle";

export class Garant {

    UUID !: String;
    cni !: String;
    dateDelivranceCni !: Date;
    nom !: String;
    prenom !: String;
    adresse !: String;
    telephone !: String;
    email !: String;
    profession !: String;
    situationProfessionnelle !: SituationProfessionnelle;
}
